airflow/providers/amazon/aws/transfers/s3_to_dynamodb.py

```diff
@@ -0,0 +1,260 @@
+                },
+                InputFormat=self.input_format,
+                TableCreationParameters={
+                    "TableName": table_name,
+                    "AttributeDefinitions": self.dynamodb_attributes,
+                    "KeySchema": self.dynamodb_key_schema,
+                    "BillingMode": self.billing_mode,
+                    **import_table_creation_config,
+                },
+                **import_table_config,
+            )
+        except ClientError as e:
+            self.log.error("Error: failed to load from S3 into DynamoDB table. Error: %s", str(e))
+            raise AirflowException(f"S3 load into DynamoDB table failed with error: {e}")
+
+        if response["ImportTableDescription"]["ImportStatus"] == "FAILED":
+            raise AirflowException(
+                "S3 into Dynamodb job creation failed. Code: "
+                f"{response['ImportTableDescription']['FailureCode']}. "
+                f"Failure: {response['ImportTableDescription']['FailureMessage']}"
+            )
+
+        if self.wait_for_completion:
+            self.log.info("Waiting for S3 into Dynamodb job to complete")
+            waiter = dynamodb_hook.get_waiter("import_table")
+            try:
+                waiter.wait(
+                    ImportArn=response["ImportTableDescription"]["ImportArn"],
+                    WaiterConfig={"Delay": self.check_interval, "MaxAttempts": self.max_attempts},
+                )
+            except WaiterError:
+                status, error_code, error_msg = dynamodb_hook.get_import_status(
+                    response["ImportTableDescription"]["ImportArn"]
+                )
+                if delete_on_error:
+                    client.delete_table(TableName=table_name)
+                raise AirflowException(
+                    f"S3 import into Dynamodb job failed: Status: {status}. Error: {error_code}. Error message: {error_msg}"
+                )
+        return response["ImportTableDescription"]["ImportArn"]
+
+    def _load_into_existing_table(self, context: Context) -> str:
+        """
+        Import S3 key or keys in an existing DynamoDB table.
+
+        :param context: The current context of the task instance
+        :return:The Amazon resource number (ARN)
+        """
+        if not self.wait_for_completion:
+            raise ValueError("wait_for_completion must be set to True when loading into an existing table")
+        table_keys = [key["AttributeName"] for key in self.dynamodb_key_schema]
+
+        dynamodb_hook = DynamoDBHook(
+            aws_conn_id=self.aws_conn_id, table_name=self.dynamodb_table_name, table_keys=table_keys
+        )
+        client = dynamodb_hook.client
+
+        self.log.info("Loading from S3 into a tmp DynamoDB table %s", self.tmp_table_name)
+        self._load_into_new_table(
+            context=context, table_name=self.tmp_table_name, delete_on_error=self.delete_on_error
```

Comment:
```suggestion
            context=context, table_name=self.tmp_table_name, delete_on_error=True
```

Reply:
The `delete_on_error` parameter controls whether a DynamoDB table is deleted in case of import errors. It is set to False by default because we want users to conset to this behavior. It arose from this [comment](https://github.com/apache/airflow/pull/39654#discussion_r1604601423). If we always set it to True, users will no longer be able to change the behavior.

Reply:
We are talking about 2 different tables:
- The table created when `use_existing_table` is `False`. Whether we remove this table in case of error should be configured by the user with `delete_on_error`
- The temporary table created when `use_existing_table` is `True`. This table, to me, is implementation details and the suer is not even aware of its existence. Thus, to me, we should always remove this table in case of error